apicast/src/backend/cache_handler.lua

```diff
@@ -17,6 +17,14 @@ local mt = {
   end,
 }
 
+-- Returns the rejection reason from the headers of a 3scale backend response.
+-- The header is set only when the authrep call to backend enables the option
+-- to get the rejection reason. This is specified in the '3scale-options'
+-- header of the request.
+local function rejection_reason(response_headers)
+  return response_headers and response_headers['3scale-rejection-reason']
```

Comment:
Should it return `'not authorized'` when there is no response for whatever reason?